### PR TITLE
Add Squiddy action for adding pull requests to Trello cards

### PR DESCRIPTION
**Pull request by surminus**
This adds everything required to enable automating creating a checklist in Trello when a Trello link is detected in a pull request.

I've created a bunch of interfaces, with the intention that they can be reusable for other actions.

### Squiddy::PullRequest

This contains some basic functionality for working with Pull Requests. I did originally consider sending the entire payload, but decided it was easier to use the Octokit library to query the pull request itself.

### TrelloChecklist

This is a very specific class to adding a checklist to a Trello card, adding items, and marking them as complete.

### Event

This automatically uses the environment variables set by the GitHub Actions environment and makes them accessible to use in other classes.

### Runner and Dockerfile

I've also added the basic structure for adding different actions inside the runner, and set up the process for creating the pre-built image.

It's worth noting that the bundle inside the Docker image must be built using `--system` due to the way GitHub Actions mounts the container when starting.

Using the Alpine image kept it pretty small, and so it can download and run relatively quickly as an action.

I've got an example test run of it working [here](https://github.com/futurelearn/test-github-actions/runs/903318587?check_suite_focus=true), and the Trello card [here](https://trello.com/c/U7tuZS2i/878-lauras-test-trello-card).

### CODE REVIEW

**Comment by surminus** [comment]
I discovered a bug when testing this manually where if a checklist doesn't exist, it throws an error, so need to add a rescue and some more tests.

But otherwise I think this is almost ready!

File: .github/workflows/spec.yaml [hunk 1]
```diff
@@ -1,11 +1,22 @@
...
     - uses: actions/checkout@v2
-    - uses: actions/setup-ruby@v1
-    - run: bundle install
+    - id: branch_name
+      run: echo ::set-output name=branch::${GITHUB_REF#refs/heads/}
+    - uses: ruby/setup-ruby@v1
+      with:
+        bundler-cache: true
     - run: bundle exec rspec
+    - uses: docker/build-push-action@v1
```
Comment (elenatanasoiu) [inline .github/workflows/spec.yaml:15]: Why are pushing an image after running specs?

Comment (surminus) [reply .github/workflows/spec.yaml:15]: It's useful for testing. It gets tagged to the branch. Is there a reason we may not wish to do this?

Comment (elenatanasoiu) [reply .github/workflows/spec.yaml:15]: Is there some sort of API limit to the amount of images we can push to docker? 

I wouldn't expect it on a spec run, but if it useful 👍

Comment (surminus) [reply .github/workflows/spec.yaml:15]: Nope, and since we're tagging it with the same tag, it'll replace the image in Docker Hub.

**Comment by surminus** [comment]
:robot:

**Pull request merged**